docs/labs/JavavsCPP.md

```diff
@@ -157,6 +157,29 @@ you created a variable that holds a pointer only,
 but you didn't create space for the Fraction object itself.
 By calling ```new```.
 
+### Functions, passing by reference and passing by value
+
+The side effect of having no pointer locations is that you cannot pass parameters into a function by
+reference in Java.
+C++ supports passing parameters both by value and reference, so you can have multiple outputs in a function.
+
+```cpp
+void myFunction(
+  int a, // in
+  int* b, // in
+  int* c, // out
+  int* d, // out
```

Comment:
what are the comments in here?
Did you take this from somewhere in particular? 
If so, we need to cite it

Reply:
It is to signify which variables are used as inputs and which variables are used as outputs. I came up with the example myself and did not take it from anywhere. Do you want me to update those comments for extra clarification?